src/coreclr/jit/optimizer.cpp

```diff
@@ -1319,6 +1319,13 @@ PhaseStatus Compiler::optUnrollLoops()
         fgDfsBlocksAndRemove();
         m_loops = FlowGraphNaturalLoops::Find(m_dfsTree);
 
+        if (optCanonicalizeLoops())
+        {
+            fgInvalidateDfsTree();
+            m_dfsTree = fgComputeDfs();
+            m_loops   = FlowGraphNaturalLoops::Find(m_dfsTree);
+        }
```

Comment:
Is this necessary for unrolling? If we've fully unrolled a loop, then the exit blocks are no longer unique (they have N preds for N unrolls), but there is no loop left. Presumably all the other loops haven't changed.

Reply:
Loop unrolling does not necessarily remove the loop; if the loop has multiple backedges then it remains a natural loop after redirecting the loop test backedge.